lib/modules/programmingLanguages.nix

```diff
@@ -0,0 +1,33 @@
+{ lib }:
+
+let
+  inherit (lib)
+    mkOption
+    types
+    ;
+in
+rec {
+  programmingLanguagesModule = types.submodule {
```

Comment:
I am a bit unclear on this,  do we need this per language or common `programmingLanguagesModule` for all languages suffice. 
Drawback of this is less per language flexibility. 
Implementation question, if we keep this common for all languages how do we update vscode `extensions` when user has enabled say both java and go (just an example).

Feel free to suggest alternatives if i am way off :)

Reply:
`programmingLanguagesModule` is global, i.e. every language will be of this type. And then inside of it you define how each module (program/service) is going to be defined, i.e. what configuration/packages it needs to implement a language.

Reply:
> Implementation question, if we keep this common for all languages how do we update vscode extensions when user has enabled say both java and go (just an example).

I assume that option is a list or a set, in which case we merge the defined list (set) for java with the one defined for go.

Reply:
cool thanks that answers it.